lib/watir/window_collection.rb

```diff
@@ -0,0 +1,110 @@
+module Watir
+  class WindowCollection
+    include Enumerable
+    include Waitable
+
+    def initialize(browser, selector = {})
+      unless selector.keys.all? { |k| %i[title url element].include? k }
+        raise ArgumentError, "invalid window selector: #{selector.inspect}"
+      end
+
+      @browser = browser
+      @selector = selector
+    end
+
+    #
+    # Yields each window in collection.
+    #
+    # @yieldparam [Watir::Window]
+    #
+
+    def each(&blk)
+      reset!
+      to_a.each(&blk)
+    end
+
+    alias length count
+    alias size count
+    alias empty? none?
+
+    #
+    # First window of the collection
+    #
+    # @note windows in a collection are not ordered so this is not reliably
+    # @deprecated use Browser#switch_window or a better Window locator
+    # @return [Watir::Window] Returns an instance of a Watir::Window
+    #
+
+    def first
+      self[0]
+    end
+
+    #
+    # Last window of the collection
+    #
+    # @note windows in a collection are not ordered so this is not reliably
+    # @deprecated use Browser#switch_window or a better Window locator
+    # @return [Watir::Window] Returns an instance of a Watir::Window
+    #
+
+    def last
+      self[-1]
+    end
+
+    #
+    # Get the element at the given index or range.
```

Comment:
Not to be picky... but still referencing "element" here.

Reply:
gah. thanks. again.